kernel/syscall.c

```diff
@@ -3715,7 +3717,18 @@ static long _syscall(void* args_)
                 }
             }
 
-            BREAK(_return(n, (long)myst_munmap(addr, length)));
+            long ret = (long)myst_munmap(addr, length);
```

Comment:
we really need to remove it from the process mapping structure. If it is a partial unmap then the process mappings need to be updated accordingly

Reply:
As noted above, this is a half-measure to workaround the crash caused by releasing memory that has since be given to another process. The full-measure will involve removing the process-mapping structures altogether (now that we have the pids[] vector). But the ``/proc/self/maps`` logic uses these mappings and needs to be reworked as well.